core/src/main/java/io/camunda/migrator/AutoDeployer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.migrator;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.client.api.command.DeployResourceCommandStep1;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+public class AutoDeployer {
```

Comment:
Nit:
This is only gonna be used for C8 deployments, right? If so I would like the name to be more specific so its clearer this isnt deploying anything else. Maybe something like `C8ProcessAutoDeployer`?

Reply:
Having this in the class name might make understanding what API it calls easier. However, I'm a bit afraid of cluttering the whole migrator code base with C7 and C8 prefixes and I'm unsure if this really provides much value. We might extend this class in the future with C7 auto-deployment capabilities. Would it mean that we have to rename this class again? What would be the name? C7C8AutoDeployer. Not sure if that is great TBH. Isn't it relatively straightforward what happens and which APIs are involved by just reading the code?

Reply:
Yeah thats fair, especially if we plan on extending it.